src/Components/SessionTimeout/WarningLogoutDialog.tsx

```diff
@@ -0,0 +1,43 @@
+import React, {useEffect, useState} from 'react';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+
+interface IWarningLogOutDialogProps {
+  openDialog: boolean;
+  getRemainingTime: () => number;
+  broadcastMessage: (data: string | number | object, emitOnSelf?: boolean | undefined) => boolean;
+}
+const WarningLogOutDialog: React.FC<IWarningLogOutDialogProps> = ({openDialog, getRemainingTime, broadcastMessage}) => {
+  const [countdown, setCountdown] = useState<number>(0);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setCountdown(Math.ceil(getRemainingTime() / 1000));
+    }, 500);
+
+    return () => {
+      clearInterval(interval);
+    };
+  });
```

Comment:
This will run on every re-render, specify some dependency in useEffect to avoid that.